src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -11958,6 +11989,12 @@ class C { }
                 """;
 
             CreateCompilation(source).VerifyEmitDiagnostics(
+                // (12,12): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                // [A<object>(B<delegate*<void>[]>.C)]
+                Diagnostic(ErrorCode.ERR_UnsafeNeeded, "B<delegate*<void>[]>").WithLocation(12, 12),
+                // (12,12): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                // [A<object>(B<delegate*<void>[]>.C)]
+                Diagnostic(ErrorCode.ERR_UnsafeNeeded, "B<delegate*<void>[]>.C").WithLocation(12, 12),
```

Comment:
The diagnostic for `delegate*` is reported in `BindNamespaceOrTypeOrAliasSymbol` (part of `BindType`).
The diagnostic for `B<delegate*<void>[]>` is reported in `BindExpression` (because it's on the left of a member access).
The diagnostic for `B<delegate*<void>[]>.C` is reported in `BindExpression`.